### PR TITLE
feat: make disallowInsecureSha1 support subspace

### PR DESCRIPTION
**Pull request by chengcyber**
## Summary

This PR makes the "disallowInsecureSha1" policy to be subspace-level.

## Details

Previously, the whole policy is enabled by the pnpm-config.json in the default subspace. This PR reads the pnpm-config.json from each subspace when validating the "pnpm-lock.yaml"

## How it was tested

Local

### CODE REVIEW

File: libraries/rush-lib/src/logic/base/BaseShrinkwrapFile.ts [hunk 1]
```diff
@@ -34,7 +34,11 @@ export abstract class BaseShrinkwrapFile {
    *
    * @virtual
    */
-  public validateShrinkwrapAfterUpdate(rushConfiguration: RushConfiguration, terminal: ITerminal): void {}
+  public validateShrinkwrapAfterUpdate(
+    rushConfiguration: RushConfiguration,
+    subspace: Subspace,
```
Comment (iclanton) [inline libraries/rush-lib/src/logic/base/BaseShrinkwrapFile.ts:39]: Should this be


```suggestion
    subspace: Subspace | undefined,
```

?

File: libraries/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts [hunk 2]
```diff
@@ -373,8 +373,13 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
   }
 
   /** @override */
-  public validateShrinkwrapAfterUpdate(rushConfiguration: RushConfiguration, terminal: ITerminal): void {
-    const { pnpmLockfilePolicies } = rushConfiguration.pnpmOptions;
+  public validateShrinkwrapAfterUpdate(
+    rushConfiguration: RushConfiguration,
+    subspace: Subspace,
```
Comment (iclanton) [inline libraries/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts:378]:
```suggestion
    subspace: Subspace | undefined,
```

File: common/changes/@microsoft/rush/feat-secure-sha1-subspace_2024-04-30-21-46.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/rush",
+      "comment": "Make disallowInsecureSha1 policy to be subspace level configuration",
```
Comment (iclanton) [inline common/changes/@microsoft/rush/feat-secure-sha1-subspace_2024-04-30-21-46.json:5]:
```suggestion
      "comment": "Make `disallowInsecureSha1` policy a subspace-level configuration.",
```

**Review by iclanton — APPROVED** [review]

File: libraries/rush-lib/src/logic/base/BaseShrinkwrapFile.ts (hunk 1 above)
Comment (octogonz) [reply libraries/rush-lib/src/logic/base/BaseShrinkwrapFile.ts:39]: I'm guessing not. The future is that Rush Subspaces will be an always-on feature, not an opt-in feature.  The opt-in part is adding more than one subspace beyond the default one.  Thus the APIs generally should NOT accept `undefined`

**Review by octogonz — APPROVED** [review]

**Pull request merged**